baseplate/context/__init__.py

```diff
@@ -51,8 +51,9 @@ def __init__(self, name, context_factory):
         self.context_factory = context_factory
 
     def on_server_span_created(self, context, server_span):
-        context_attr = self.context_factory.make_object_for_context(self.name, server_span)
-        setattr(context, self.name, context_attr)
+        def context_attr(_self):
+            return self.context_factory.make_object_for_context(self.name, server_span)
+        context.add_lazy_attribute(self.name, context_attr)
```

Comment:
Would it make sense to make this optional?  It could default to on but give services the option to turn it off?  I could see it being the case that this helps some services but hurts others.

Reply:
Maybe a `LazyContextFactory` and `EagerContextFactory` and then alias the old `ContextFactory` to whichever profiles out to be the more common-case friendly one? Then services can decide based on the individual context attributes based on their knowledge of their own usage patterns.